### PR TITLE
executor: change log level to INFO when fetching next chunk failed

### LINKED ISSUE #45951: The log level when fetching next chunk failed is a bit high
## Enhancement
When fetching next chunk failed, it may be caused by the server canceled the SQL execution, and it's  message will finally send to the client, so it could not be ignored. But the current log level of it is ERROR, it is a bit too high, which makes some users concern about it.

### PR DESCRIPTION
**Pull request by jackysp**
### What problem does this PR solve?

Issue Number: close #45951

Problem Summary:

### What is changed and how it works?
Change the log level to INFO.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: executor/distsql.go [hunk 1]
```diff
@@ -1492,7 +1492,7 @@ func (w *tableWorker) executeTask(ctx context.Context, task *lookupTableTask) er
 		chk := tryNewCacheChunk(tableReader)
 		err = Next(ctx, tableReader, chk)
 		if err != nil {
-			logutil.Logger(ctx).Error("table reader fetch next chunk failed", zap.Error(err))
+			logutil.Logger(ctx).Info("table reader fetch next chunk failed", zap.Error(err))
```
Comment (jackysp) [inline executor/distsql.go:1495]: It seems it's hard to reach this line in the unit tests.

Comment (hawkingrei) [reply executor/distsql.go:1495]: Now tidb is in go1.21. in Go 1.21, The new [AfterFunc](https://tip.golang.org/pkg/context/#AfterFunc) function registers a function to run after a context has been cancelled. so maybe we check it to set log level.

Comment (hawkingrei) [reply executor/distsql.go:1495]: WithCancelCause will be better?

Comment (jackysp) [reply executor/distsql.go:1495]: I found that sometimes this log still has unique value. It feels better to keep it as it is.

**Pull request closed without merging**